setup.py

```diff
@@ -25,38 +29,59 @@ def package_files(directory):
     "../mlflow/temporary_db_migrations_for_pre_1_users/alembic.ini",
 ]
 
+"""
+Minimal requirements for the skinny MLflow client which provides a limited
+subset of functionality such as: RESTful client functionality for Tracking and
+Model Registry, as well as support for Project execution against local backends
+and Databricks.
+"""
+SKINNY_REQUIREMENTS = [
+    "click>=7.0",
+    "cloudpickle",
+    "databricks-cli>=0.8.7",
+    "entrypoints",
+    "gitpython>=2.1.0",
+    "numpy",
+    "pandas",
+    "python-dateutil",
+    "pyyaml",
+    "protobuf>=3.6.0",
+    "requests>=2.17.3",
+    "six>=1.10.0",
+    "simplejson",
+]
+
+_is_mlflow_skinny = bool(os.environ.get(_MLFLOW_SKINNY_ENV_VAR))
+logging.debug("{} env var is set: {}".format(_MLFLOW_SKINNY_ENV_VAR, _is_mlflow_skinny))
+
+"""
+These are the core requirements for the complete MLflow platform, which augments
+the skinny client functionality with support for running the MLflow Tracking
+Server & UI. It also adds project backends such as Docker and Kubernetes among
+other capabilities.
+"""
+CORE_REQUIREMENTS = SKINNY_REQUIREMENTS + [
+    "alembic<=1.4.1",
+    # Required
+    "azure-storage-blob",
+    "docker>=4.0.0",
+    "Flask",
+    "gunicorn; platform_system != 'Windows'",
+    "prometheus-flask-exporter",
+    "querystring_parser",
+    # Pin sqlparse for: https://github.com/mlflow/mlflow/issues/3433
+    "sqlparse>=0.3.1",
+    # Required to run the MLflow server against SQL-backed storage
+    "sqlalchemy<=1.3.13",
+    "waitress; platform_system == 'Windows'",
+]
+
 setup(
     name="mlflow",
     version=version,
     packages=find_packages(exclude=["tests", "tests.*"]),
     package_data={"mlflow": js_files + models_container_server_files + alembic_files},
```

Comment:
I don't think we should embed the UI JS, model container server files, or alembic files with the skinny client. Can we condition adding these files on `_is_mlflow_skinny` and omit them if skinny is True?

Reply:
Hmm...looks like these files are included independently of `package_data`. I think `find_packages` picks them up. Let's follow up here in a future PR.